services/worker-manager/src/providers/aws.js

```diff
@@ -62,24 +62,26 @@ class AwsProvider extends Provider {
       });
     }
 
-    const toSpawn = await this.estimator.simple({
+    let toSpawn = await this.estimator.simple({
       workerPoolId,
       minCapacity: workerPool.config.minCapacity,
       maxCapacity: workerPool.config.maxCapacity,
-      capacityPerInstance: 1, // todo: this will be corrected along with estimator changes (bug 1579554)
-      running: workerPool.providerData[this.providerId].running,
+      runningCapacity: workerPool.providerData[this.providerId].running,
     });
     if (toSpawn === 0) {
       return;
     }
-    const toSpawnPerConfig = Math.ceil(toSpawn / workerPool.config.launchConfigs.length);
 
-    const shuffledConfigs = _.shuffle(workerPool.config.launchConfigs);
+    const cfgs = [];
+    const maxPerConfig = 15;
+    while (toSpawn> 0) {
+      const cfg = _.sample(workerPool.config.launchConfigs);
+      const provided = Math.min(toSpawn, maxPerConfig * cfg.capacityPerInstance);
+      cfgs.push([provided, cfg]);
+      toSpawn -= provided;
+    }
```

Comment:
What is the reason for this change? What was wrong with the previous logic?

Reply:
The old way could not handle different `capacityPerInstance` for different `launchConfigs`.